### PR TITLE
MAISTRA-2389: Ensure the status code can handle Forbidden errors

### PR DESCRIPTION
**Pull request by knrc**
Fix status update code to handle Forbidden errors.

### CODE REVIEW

File: pilot/pkg/status/state.go [hunk 1]
```diff
@@ -174,8 +174,8 @@ func (c *DistributionController) writeStatus(ctx context.Context, config Resourc
 	// should this be moved to some sort of InformerCache for speed?
 	current, err := resourceInterface.Get(ctx, config.Name, metav1.GetOptions{ResourceVersion: config.ResourceVersion})
 	if err != nil {
-		if errors.IsGone(err) || errors.IsNotFound(err) {
-			// this resource has been deleted.  prune its state and move on.
+		if errors.IsGone(err) || errors.IsNotFound(err) || errors.IsForbidden(err) {
```
Comment (rcernich) [inline pilot/pkg/status/state.go:177]: I think this works around the problem, but should it be happening at all if the controller has the correct list of members?

Comment (knrc) [reply pilot/pkg/status/state.go:177]: It has nothing to do with the controller, this is going direct to the API server which is why we get the problem

Comment (knrc) [reply pilot/pkg/status/state.go:177]: Note the way this works in that there is a configmap created in the control plane namespace containing status updates for resources which have been propagated, the leader then updates those resources directly by going straight to the API server.  There is no integration with SMMR updates.

Comment (knrc) [reply pilot/pkg/status/state.go:177]: I also checked other parts of the codebase to see if this was used elsewhere but could not find anything.

**Review by rcernich — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

**Pull request merged**